File: sod/sod-monitor-center/sod-monitor-server/src/main/java/com/piesat/monitor/config/MybatisPlusConfigGeneral.java
package com.piesat.monitor.config;

import org.apache.ibatis.session.SqlSessionFactory;
import org.mybatis.spring.SqlSessionFactoryBean;
import org.mybatis.spring.SqlSessionTemplate;
import org.mybatis.spring.annotation.MapperScan;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
import org.springframework.jdbc.datasource.DataSourceTransactionManager;

import javax.sql.DataSource;


/**
 * @program: sod
 * @description:
 * @author: zzj
 * @create: 2020-03-19 10:11
 **/
@Configuration
@MapperScan(basePackages = "com.piesat.monitor.dao.general", sqlSessionTemplateRef = "generalSqlSessionTemplate")
public class MybatisPlusConfigGeneral {
    //general数据源
    @Bean("generalSqlSessionFactory")
    public SqlSessionFactory generalSqlSessionFactory(@Qualifier("generalDataSource") DataSource dataSource) throws Exception {
        SqlSessionFactoryBean sqlSessionFactory = new SqlSessionFactoryBean();
        sqlSessionFactory.setDataSource(dataSource);
        sqlSessionFactory.setMapperLocations(new PathMatchingResourcePatternResolver().
                getResources("classpath*:general/*.xml"));
        return sqlSessionFactory.getObject();
    }

    //事务支持
    @Bean(name = "generalTransactionManager")
    public DataSourceTransactionManager generalTransactionManager(@Qualifier("generalDataSource") DataSource dataSource) {
        return new DataSourceTransactionManager(dataSource);
    }

    @Bean(name = "generalSqlSessionTemplate")
    public SqlSessionTemplate generalSqlSessionTemplate(@Qualifier("generalSqlSessionFactory") SqlSessionFactory sqlSessionFactory) {
        return new SqlSessionTemplate(sqlSessionFactory);
    }
}

File: sod/sod-monitor-center/sod-monitor-server/src/main/scala/com/piesat/monitor/config/MybatisPlusConfigGeneral.scala
package com.piesat.monitor.config


import javax.sql.DataSource

import com.alibaba.druid.pool.DruidDataSource
import org.apache.ibatis.session.SqlSessionFactory
import org.mybatis.spring.annotation.MapperScan
import org.mybatis.spring.boot.autoconfigure.MybatisProperties
import org.mybatis.spring.{SqlSessionFactoryBean, SqlSessionTemplate}
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.jdbc.DataSourceBuilder
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.core.io.Resource
import org.springframework.core.io.support.PathMatchingResourcePatternResolver
import org.springframework.jdbc.datasource.DataSourceTransactionManager

/**
  * Created by zzj on 2020/3/19.
  */
//@Configuration
//@MapperScan(basePackages = Array("com.piesat.monitor.dao.general"), sqlSessionTemplateRef = "generalSqlSessionTemplate")
class MybatisPlusConfigGeneral { //general数据源

  @Bean(name = Array("generalDataSource"))
  @ConfigurationProperties(prefix = "spring.datasource.general")
  def  generalDataSource() :DataSource={
    return new DruidDataSource;
  }
  @Bean(Array("generalSqlSessionFactory"))
  @throws[Exception]
  def generalSqlSessionFactory(@Qualifier("generalDataSource") dataSource: DataSource): SqlSessionFactory = {
    val sqlSessionFactory = new SqlSessionFactoryBean
    sqlSessionFactory.setDataSource(dataSource)
    var resource:Array[Resource]= new PathMatchingResourcePatternResolver().getResources("classpath*:general/**.xml")
    //sqlSessionFactory.setConfiguration(mybatisProperties.getConfiguration)
    sqlSessionFactory.setMapperLocations(resource:_*)
    sqlSessionFactory.getObject
  }

  //事务支持
  @Bean(name = Array("generalTransactionManager")) def generalTransactionManager(@Qualifier("generalDataSource") dataSource: DataSource) = new DataSourceTransactionManager(dataSource)

  @Bean(name = Array("generalSqlSessionTemplate")) def generalSqlSessionTemplate(@Qualifier("generalSqlSessionFactory") sqlSessionFactory: SqlSessionFactory) = new SqlSessionTemplate(sqlSessionFactory)
}

